ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundleForm.tsx

```diff
@@ -0,0 +1,235 @@
+const nameValidatorRegExp = /^[a-zA-Z0-9._-]+$/;
+
+const validationSchema: yup.ObjectSchema<InitBundleFormValues> = yup.object().shape({
+    name: yup
+        .string()
+        .trim()
+        .matches(
+            nameValidatorRegExp,
+            'Name can have only the following characters: letters, digits, period, underscore, hyphen (but no spaces)'
+        )
+        .required('Bundle name is required'),
+    installation: yup.string().trim().required(), // Select
+    platform: yup.string().trim().required(), // Radio
+});
+
+function InitBundleForm(): ReactElement {
+    const history = useHistory();
+    const [errorMessage, setErrorMessage] = useState('');
+    const {
+        errors,
+        handleBlur,
+        isSubmitting,
+        isValid,
+        setFieldValue,
+        setValues,
+        submitForm,
+        touched,
+        values,
+    } = useFormik({
+        initialValues,
+        onSubmit: (values, { setSubmitting }) => {
+            setSubmitting(true);
+            const { installation, name } = values;
+            generateClusterInitBundle({ name })
+                .then(({ response }) => {
+                    setErrorMessage('');
+                    downloadBundle(installation, response); // TODO try catch?
+                    setSubmitting(false);
+                    goBack();
+                })
+                .catch((error) => {
+                    setErrorMessage(getAxiosErrorMessage(error));
+                    setSubmitting(false);
+                });
+        },
+        validateOnMount: true, // disable Next when Name is empty
+        validationSchema,
+    });
+    const { isOpen, onToggle } = useSelectToggle();
+
+    function goBack() {
+        history.goBack(); // to InputBundlesTable or NoClustersPage
+    }
+
+    // return setWhatever solves problem reported by typescript-eslint no-floating-promises
+
+    function onChangeTextInput(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    function onChangePlatform(value) {
+        return setValues({
+            installation: value === 'OpenShift' ? 'Operator' : 'Helm',
+            name: values.name, // redundant but function requires all values
+            platform: value,
+        });
+    }
+
+    function onSelectInstallation(value) {
+        return setFieldValue('installation', value);
+    }
+
+    return (
+        <>
+            <InitBundlesHeader title="Create bundle" />
+            <Divider component="div" />
+            <PageSection variant="light">
+                <Flex direction={{ default: 'column' }}>
+                    <Form>
+                        <FormLabelGroup
+                            fieldId="name"
+                            label="Name"
+                            isRequired
+                            errors={errors}
+                            touched={touched}
+                        >
```

Comment:
Suggestion for future PR:
It would be nice to show the requirements for this value in the PF component's `helperText`, rather than waiting for an error to show the user.
```suggestion
                            errors={errors}
                            touched={touched}
                            helperText="Name can have only the following characters: letters, digits, period, underscore, hyphen (but no spaces)"
                        >
```

Reply:
Super suggestion!